SIPS/sip-16.md

```diff
@@ -0,0 +1,209 @@
+Instead, it is the Snap developer's responsibility to be cognizant of the shape of signature objects for relevant chains.
+Nevertheless, you can refer to [Appendix I](#appendix-i-ethereum-signature-objects) for the interfaces of the Ethereum signature objects available in MetaMask at the time of this SIP's creation.
+
+`chainId` - This is a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) `chainId` string.
+The snap is expected to parse and utilize this string as needed.
+
+`signatureOrigin` - The URL origin of the signature request. The existence of this property is dependent on the `allowSignatureOrigin` caveat.
+
+
+The interface for the return value of an `onSignature` export is:
+
+```typescript
+interface OnSignatureResponse {
+  content: Component | null;
+  severity?: SeverityLevel;
+}
+```
+
+**Note:** `severity` is an OPTIONAL field and the omission of such means that there is no escalation of the content being returned.
+
+## Specification
+
+Please see [SIP-3](sip-3.md) for more information on the original transaction insights API.
+
+Please see [SIP-7](sip-7.md) for more information on the `Component` type returned in the `OnTransactionResponse`.
+
+Please see [SIP-11](sip-3.md) for more information on Transaction insight severity levels.
+
+## Appendix I: Ethereum Signature Objects
+
+The following signature objects may appear for any `chainId` of `eip155:*` where `*` is some positive integer. This includes all Ethereum or "EVM-compatible" chains. As of the time of the creation of this SIP, they are the only possible signature objects for Ethereum chains.
+
+### eth_sign
+
+```typescript
+interface EthSignatureObj {
+  from: string;
+  data: string;
+  metamaskId: string;
+  signatureMethod: 'eth_sign';
+}
+```
+
+### personal_sign
+
+```typescript
+interface PersonalSignatureObj {
+  from: string;
+  data: string;
+  metamaskId: string;
+  signatureMethod: 'personal_sign';
+}
+```
+
+### eth_signTypedData
+
+```typescript
+interface SignTypedDataSignatureObj {
+  from: string;
+  data: Record<string, any>;
+  metamaskId: string;
+  signatureMethod: 'eth_signTypedData';
+}
+```
+
+### eth_signTypedData_v3
+
+```typescript
+interface SignTypedDataV3SignatureObj {
+  from: string;
+  data: Record<string, any>;
+  metamaskId: string;
+  signatureMethod: 'eth_signTypedData_v3';
+  version: 'V3';
+}
+```
+
+### eth_signTypedData_v4
+
+```typescript
+interface SignTypedDataV4SignatureObj {
+  from: string;
+  data: Record<string, any>;
+  metamaskId: string;
+  signatureMethod: 'eth_signTypedData_v4';
+  version: 'V4';
+}
+```
+
+**Note**: The `metamaskId`, `signatureMethod` and `version` properties are MetaMask specific properties and not reflective of the standards defining the underlying signature methods. `signatureMethod` SHOULD be used by the signature insight snap as the source of the truth to identify the signature scheme it is providing insights for.
```

Comment:
Wondering if we should strip `metamaskId` and `version` from the payload? It seems like noise that no one will need?

Reply:
Hmm yeah, I suppose so.